en/docs/setup/migrating-preparing-for-migration.md

```diff
@@ -21,6 +21,113 @@
+* Any custom external claim mapped to the wso2.role claim should be mapped to either wso2.roles 
+  or wso2.groups claim as per the requirement of the custom use case.
+* Our recommendation is to fix any consuming client to become compatible with these changes. 
+  But if somehow the above configs need to be reverted(possibly in a migrated deployment), it can 
+  be done via calling IS server REST APIs(https://is.docs.wso2.com/en/latest/develop/claim-management-rest-api/).
+  
+### Utilizing carbon kernel level support
+
+The following abstract userstore manager APIs: getUserClaimValues(), getUsersClaimValues(), 
+getUserClaimValuesWithID(), getUsersClaimValuesWithID() now support both wso2.roles 
+and wso2.groups claims properly. We recommend modifying custom extensions to request wso2.roles or wso2.groups
+via above APIs rather depending on wso2.role claim.
+
+### Service provider role mapping, and IdP role mapping  are restricted strictly to role mapping
+
+We recommend removing existing SP and IdP role mappings which use groups, and utilize roles to achieve the same functionality.
+
+### OIDC group claim return groups
+
+OIDC group claim does not return internal roles anymore. We recommend modifying applications and  
+custom extensions to utilize this behaviour. If roles are required, utilize the OIDC roles claim.
+
+### Obtaining roles via the SAML assertion
+
+We recommend applications and custom extensions to switch from wso2.role to the wso2.roles claim in 
+the SAML assertion.
+
+### SCIM2 roles.default claim returns roles and groups claim return groups
+
+Previously the roles.default claim in SCIM2 returned both groups and roles as it was mapped to 
+the wso2.roles claim. Going forward it is mapped to the wso2.roles claim, where only roles are 
+returned. In order to get groups, `urn:ietf:params:scim:schemas:core:2.0:User:groups` claim should 
+be used instead since with this improvement it is returning groups as intended.
+
+### Groups and roles in SCIM2 user response
+
+Previously users and roles in the SCIM2 user response returned as a comma separated single entity. 
+However, that has been changed, and now they return as separate complex entities. We recommend 
+modifying clients that consume this response.
+
+### Backward compatibility
+
+Groups and roles separation improvements brings enhanced clarity, and improved performance to the 
+product. However, it’s inevitable to bring all the goodness with zero compromises. Therefore, as 
+mentioned above, some applications, customizations, and integration flows might need some changes 
+to fully adapt to these improvements.
+
+Nevertheless, all of the above improvements and the behavioural changes are introduced in a way 
+that existing deployments can adapt to the new state as easy as possible. However, if it's 
+mandatory to preserve previous behaviour and avoid enabling the improvements mentioned above, the 
+following configuration option(enabled by default) can be used.
```

Comment:
better to mention the configuration file name and the location

Reply:
Fixed with fa6da06b8ecc8fd6ac34122f47ef8e9050296214